src/common/templates/_storage.tpl

```diff
@@ -19,3 +19,36 @@
   {{- end -}}
 {{- end -}}
 {{- end -}}
+{{/*
+Define a list of volumes for a pod.
+
+Usage:
+{{ include "harnesscommon.volumes" (dict "volumes" (list (dict "name" "harness-opt") (dict "name" "tmp")) ) }}
+
+Params:
+  - volumes: List - Required. List of volume definitions, each as a dict:
+      - name: String - Required.
+*/}}
+{{- define "harnesscommon.volumes" -}}
+{{- range .volumes }}
+- name: {{ .name }}
+  emptyDir: {}
+{{- end }}
+{{- end }}
+{{/*
+Define a list of volumeMounts for a container.
+
+Usage:
+{{ include "harnesscommon.volumeMounts" (dict "mounts" (list (dict "name" "harness-opt" "mountPath" "/opt/harness") (dict "name" "tmp" "mountPath" "/tmp")) ) }}
```

Comment:
This as well, you have defined the common function for the init container, volumeMounts are part of the containers spec, so you can add it in that function itself, I do not see any benefit of this.